opamp/observiq/observiq_client.go

```diff
@@ -0,0 +1,195 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package observiq contains OpAmp structures compatible with the observiq client
+package observiq
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/observiq/observiq-otel-collector/collector"
+	"github.com/observiq/observiq-otel-collector/opamp"
+	"github.com/open-telemetry/opamp-go/client"
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"go.uber.org/zap"
+)
+
+var (
+	// ErrUnsupportedURL is error returned when creating a client with an unsupported URL scheme
+	ErrUnsupportedURL = errors.New("unsupported URL")
+)
+
+// Ensure interface is satisfied
+var _ opamp.Client = (*Client)(nil)
+
+// Client represents a client that is connected to Iris via OpAmp
+type Client struct {
+	opampClient   client.OpAMPClient
+	logger        *zap.SugaredLogger
+	ident         *identity
+	configManager opamp.ConfigManager
+	collector     collector.Collector
+
+	currentConfig opamp.Config
```

Comment:
just off the bat wondering why the config from the configManager couldn't be used in place of this?

Reply:
nevermind, i guess it would be a pain to have to parse it into the struct every time you wanted to reference it

Reply:
Yeah, config manager is meant to manage the file and contents on disk. I'm added this reference mostly for hot reloading.